tests/builtins/test_filter.py

```diff
@@ -1,53 +1,22 @@
 from .. utils import TranspileTestCase, BuiltinTwoargFunctionTestCase
 
-from unittest import expectedFailure
-
 
 class FilterTests(TranspileTestCase):
-    base_code = """
-            #placeholder while list()s etc aren't fully implemented
-            class ListLike:
-                x = %s
-                index = 0
-
-                def __iter__(self):
-                    return self
-
-                def __next__(self):
-                    self.index = self.index + 1
-                    if self.index > len(self.x):
-                        raise StopIteration
-                    return self.x[self.index]
-
-            def testish(x):
-                return %s
-
-            print(filter(testish, ListLike()))
-            mylist = ListLike()
-            print(filter(testish, mylist).__next__())
-            print(filter(testish, mylist).__next__())
-            print(filter(testish, mylist).__next__())
-            try:
-                print(filter(testish, mylist).__next__())
-            except StopIteration:
-                pass
```

Comment:
Note to other maintainers: this was dead code (and it was broken too), the tests were marked as expected failure and they were failing for the wrong reasons.
We decided to go with new tests that would focus more on the different argument combinations than in different types and predicates, and rely on the already implemented types.